### PR TITLE
[vulkan] Add image format qualifier to glsl files

### PR DESCRIPTION
**Pull request by SS-JIA**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #69332 Allow build_android.sh to forward Vulkan args
* #69331 [vulkan] Disable shader optimization to avoid Validation Errors
* **#69330 [vulkan] Add image format qualifier to glsl files**

---

## Context

Previously, our shader files did not declare any [image format qualifiers](https://www.khronos.org/opengl/wiki/Layout_Qualifier_(GLSL)#Image_formats) for image layouts. This causes the SPIR-V modules produced to declare the [StorageImageWriteWithoutFormat](https://www.khronos.org/registry/SPIR-V/specs/unified1/SPIRV.html#_a_id_capability_a_capability) capability, which requires `shaderStorageImageWriteWithoutFormat` to be enabled in [VkPhysicalDeviceFeatures](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VkPhysicalDeviceFeatures.html). `shaderStorageImageWriteWithoutFormat` is not available on some devices, causing errors to be reported by the Vulkan validation layer.

## Changes

Vulkan shaders now declare the image format explicitly so that the SPIR-V modules produced are compatible with devices that do not have `shaderStorageImageWriteWithoutFormat` enabled.

Differential Revision: [D32840909](https://our.internmc.facebook.com/intern/diff/D32840909)

### CODE REVIEW

File: cmake/VulkanCodegen.cmake [hunk 1]
```diff
@@ -9,6 +9,9 @@ set(VULKAN_GEN_ARG_ENV "")
 if(USE_VULKAN_RELAXED_PRECISION)
   string(APPEND VULKAN_GEN_ARG_ENV "precision=mediump")
 endif()
+if (USE_VULKAN_FP16_INFERENCE)
+  string(APPEND VULKAN_GEN_ARG_ENV "format=rgba16f")
```
Comment (beback4u) [inline cmake/VulkanCodegen.cmake:13]: Should we update BUCK file, too?

**Review by beback4u — APPROVED** [review]
LGTM. You will update BUCK file in a separate diff, too.

**Comment by SS-JIA** [comment]
@SS-JIA has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32840909).

File: cmake/VulkanCodegen.cmake (hunk 1 above)
Comment (SS-JIA) [reply cmake/VulkanCodegen.cmake:13]: yep, this will be updated in a separate diff.

**Comment by SS-JIA** [comment]
@SS-JIA has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32840909).

**Comment by SS-JIA** [comment]
@SS-JIA has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32840909).

**Pull request closed without merging**